txpool/txpool.go

```diff
@@ -346,63 +345,48 @@ func (p *TxPool) Pop(tx *types.Transaction) {
 	}
 }
 
-// Drop removes the given (unrecoverable) transaction from
-// its associated promoted queue (account) and rolls
-// back the account's nextNonce.
-// Will update executables with the next primary
-// from that account (if any).
+// Drop clears the entire account associated with the given transaction
+// and reverts its next (expected) nonce.
 func (p *TxPool) Drop(tx *types.Transaction) {
+	// drop the entire account and revert nonce
 	account := p.accounts.get(tx.From)
 
 	account.promoted.lock(true)
-	defer account.promoted.unlock()
+	account.enqueued.lock(true)
 
-	// pop the top most promoted tx
-	account.promoted.pop()
+	defer func() {
+		account.enqueued.unlock()
+		account.promoted.unlock()
+	}()
 
-	// update state
-	p.index.remove(tx)
-	p.gauge.decrease(slotsRequired(tx))
+	p.logger.Debug("dropping tx (unrecoverable)", "hash", tx.Hash.String())
```

Comment:
i don't think we have a notion of recoverable/unrecoverable really here.

if we tomorrow decide to use the Drop method in some other workflow, this log will be very misleading

Reply:
[Removed](https://github.com/0xPolygon/polygon-edge/pull/380#:~:text=fix%20Drop()%20method-,7a9d85a,-Merge%20state)